src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -126,18 +126,17 @@ private static bool ApplyChangesFromTC31(JObject sarifLog)
             {
                 foreach (JObject run in runs)
                 {
+                    // https://github.com/oasis-tcs/sarif-spec/issues/311
+                    modifiedLog |= MoveRulesMetadataAndConfiguration(run);
+
                     if (run["results"] is JArray results)
                     {
                         foreach (JObject result in results)
                         {
-                            string baselineState = (string)result["baselineState"];
-
-                            if ("existing".Equals(baselineState))
-                            {
-                                result["baselineState"] = "unchanged";
-                                modifiedLog = true;
-                            }
+                            // https://github.com/oasis-tcs/sarif-spec/issues/312
+                            modifiedLog |= UpdateBaselineExistingStateToUnchanged(result);
 
+                            // https://github.com/oasis-tcs/sarif-spec/issues/317
                             modifiedLog |= SetResultKindAndFailureLevel(result);
```

Comment:
>SetResultKindAndFailureLevel [](start = 43, length = 28)

So nice! isolating each issue-related change into a named method, even if it's only a couple of lines.